webplatformcompat/renderers.py

```diff
@@ -1,244 +1,242 @@
-
-        many = hasattr(field, 'child_relation')
-        model = self.model_from_obj(field)
-        if model:
-            resource_type = self.model_to_resource_type(model)
-
-            format_kwargs = {
-                'model_name': model._meta.object_name.lower()
-            }
-            view_name = '%(model_name)s-detail' % format_kwargs
-
-            links[field_name] = self.dict_class((
-                ("type", resource_type),
-                ("href", self.url_to_template(view_name, request, field_name)),
-            ))
-
-        assert field_name in resource
-        if many:
-            pks = resource[field_name]
+        response = renderer_context.get('response')
+        request = renderer_context.get('request')
+        fields_extra = renderer_context.get('fields_extra')
+        status_code = response and response.status_code
+        is_err = is_client_error(status_code) or is_server_error(status_code)
+        if data is None:
+            converted = None
+        elif is_err:
+            converted = self.convert_error(data, status_code)
+        elif all([key in data for key in self.PAGINATION_KEYS]):
+            converted = self.convert_paginated(data, request)
+        elif request and request.method == 'OPTIONS':
+            converted = {'meta': data}
         else:
-            pks = [resource[field_name]]
-
-        link_data = []
-        for pk in pks:
-            if pk is None:
-                link_data.append(None)
+            main_resource = fields_extra['id']['resource']
+            converted = self.convert_standard(
+                data, fields_extra, main_resource, request)
+
+        renderer_context["indent"] = 4
+        return super(JsonApiRC1Renderer, self).render(
+            data=converted, renderer_context=renderer_context)
+
+    def convert_error(self, data, status_code):
+        """Convert error responses to JSON API RC1 format
+
+        Error responses are dictionaries. Simple Django errors, like permision
+        denied errors, have a single key 'detail'. Field validation errors
+        have the field name as the key, and a list of error strings. Errors on
+        nested resources appear under the '_view_extra' key.
+
+        JSON API RC1 specifies that errors appear as a dictionary with the
+        key 'errors', and the value as a list of error dictionaries, such as:
+        {"errors": [
+            {"detail": "Error string",
+             "path": "/attribute",
+             "status_code": 400
+            }]
+        }
+        """
+        errors = []
+        for name, value in data.items():
+            if name == 'detail':
+                fmt_error = self.dict_class((
+                    ('detail', value),
+                    ('status', str(status_code)),
+                ))
+                errors.append(fmt_error)
+            elif name == '_view_extra':
+                for rname, error_dict in value.items():
+                    assert rname != 'meta'
```

Comment:
I tend to add a comment to asserts otherwise I forget what the deal is when I'm looking at the traceback. Plus it makes it easier if for some reason the line numbers don't line up because the code has moved around.

Maybe something like:

```
assert rname != 'meta', 'OMG! The code is on fire! Run for your lives!'
```